### PR TITLE
Discussion and posts branch

### PR DESCRIPTION
**Pull request by matdon90**
Following functionalities with tests were implemented:

-     Logged user can create discussion.
-     Discussion creator can edit discussion.
-     Administrator can open/close discussion.
-     Administrator can pin/unpin discussion.
-     Administrator can delete discussion.
-     Logged user can get list of discussions.
-     Logged user can get list of posts of chosen discussion.

-     Logged user can add posts to discussion.
-     Post's creator can delete post.
-     Post's creator can edit post.
-     Administrator can pin/unpin post.

### CODE REVIEW

File: SK.Application/Discussions/Commands/EditDiscussion/EditDiscussionCommandHandler.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+            discussionToFind.Title = request.Title ?? discussionToFind.Title;
+            discussionToFind.Description = request.Description ?? discussionToFind.Description;
+
+            var success = await _context.SaveChangesAsync(cancellationToken) > 0;
+            if (success)
+            {
+                return Unit.Value;
+            }
+            throw new RestException(HttpStatusCode.BadRequest, new { Discussion = "Problem saving changes" });
```
Comment (UgzSourceCode) [inline SK.Application/Discussions/Commands/EditDiscussion/EditDiscussionCommandHandler.cs:32]: you have this same text in DeleteDiscussionCommandHandler.cs  file, so if you move this text to resources you can use this same text too many times

File: SK.Application/Discussions/Commands/EditDiscussion/EditDiscussionCommandValidator.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+namespace SK.Application.Discussions.Commands.EditDiscussion
+{
+    public class EditDiscussionCommandValidator : AbstractValidator<EditDiscussionCommand>
+    {
+        public EditDiscussionCommandValidator()
+        {
+            RuleFor(x => x.Title)
+                .NotEmpty().WithMessage("Discussion title is required.")
+                .Length(1, 70).WithMessage("Discussion title must be at least 1 and at max 70 characters long.");
```
Comment (UgzSourceCode) [inline SK.Application/Discussions/Commands/EditDiscussion/EditDiscussionCommandValidator.cs:11]: Move text to resources ;)

File: SK.Application.IntegrationTests/Posts/Commands/PinPostTestcs.cs [hunk 3]
```diff
@@ -0,0 +1,90 @@
+﻿using Bogus;
```
[the remark this answers is not in the document]
Comment (matdon90) [reply SK.Application.IntegrationTests/Posts/Commands/PinPostTestcs.cs:1]: Done.

File: SK.Application/Discussions/Commands/EditDiscussion/EditDiscussionCommandHandler.cs (hunk 1 above)
Comment (matdon90) [reply SK.Application/Discussions/Commands/EditDiscussion/EditDiscussionCommandHandler.cs]: Added all strings to resources.

File: SK.Application/Discussions/Commands/EditDiscussion/EditDiscussionCommandValidator.cs (hunk 2 above)
Comment (matdon90) [reply SK.Application/Discussions/Commands/EditDiscussion/EditDiscussionCommandValidator.cs]: Added all strings to resources.

File: SK.Application/Discussions/Commands/DeleteDiscussion/DeleteDiscussionCommandHandler.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+            _context.Discussions.Remove(discussionToDelete);
+
+            var success = await _context.SaveChangesAsync(cancellationToken) > 0;
+
+            if (success)
+            {
+                return Unit.Value;
+            }
+            throw new RestException(HttpStatusCode.BadRequest, new { Discussion = "Problem saving changes" });
```
[the remark this answers is not in the document]
Comment (matdon90) [reply SK.Application/Discussions/Commands/DeleteDiscussion/DeleteDiscussionCommandHandler.cs]: Added all strings to resources.

**Pull request merged**